packages/frontend/app/components/reports/curriculum/learner-groups.js

```diff
@@ -0,0 +1,157 @@
+        const offeringLearnerGroups = s.offerings.map((o) => o.learnerGroups.map((g) => g)).flat();
+        const ilmLearnerGroups = s.ilmSession?.learnerGroups.map((g) => g) ?? [];
+        const learnerGroups = [...offeringLearnerGroups, ...ilmLearnerGroups];
+
+        s.learnerGroups = uniqueById(learnerGroups)
+          .map(({ title }) => title)
+          .sort();
+        return s;
+      });
+
+      return c;
+    });
+  }
+
+  get summary() {
+    return this.reportWithLearnerGroups.map((c) => {
+      return {
+        courseId: c.id,
+        courseTitle: c.title,
+        sessionCount: c.sessions.length,
+        learnerGroupsCount: this.reporting.countUniqueValuesInArray(c.sessions, 'learnerGroups'),
+        instructorsCount: this.reporting.countUniqueValuesInArray(c.sessions, 'instructors'),
+      };
+    });
+  }
+
+  get results() {
+    const origin = window.location.origin;
+    return this.reportWithLearnerGroups.reduce((acc, c) => {
+      c.sessions.forEach((s) => {
+        const path = this.router.urlFor('session', c.id, s.id);
+        let firstOfferingDate;
+        const firstOffering = s.offerings.sort(
+          (a, b) => DateTime.fromISO(a.startDate) - DateTime.fromISO(b.startDate),
+        )[0];
+        if (firstOffering) {
+          firstOfferingDate = firstOffering.startDate;
+        } else if (s.ilmSession) {
+          firstOfferingDate = s.ilmSession.dueDate;
+        }
+        s.learnerGroups.forEach((title) => {
+          acc.push({
+            courseId: c.id,
+            courseTitle: c.title,
+            year: c.year,
+            sessionTitle: s.title,
+            firstOfferingDate: firstOfferingDate
+              ? DateTime.fromISO(firstOfferingDate).toLocaleString(this.intl.primarlyLocale)
+              : '',
+            instructors: s.instructors,
+            title,
+            link: `${origin}${path}`,
+          });
+        });
+      });
+      return acc;
+    }, []);
+  }
+
+  get sortedResults() {
+    return this.results.sort(this.sortResults);
+  }
+
+  sortResults = (a, b) => {
+    if (a.courseTitle !== b.courseTitle) {
+      return a.courseTitle.localeCompare(b.courseTitle);
+    }
+
+    return a.sessionTitle.localeCompare(b.sessionTitle);
+  };
+
+  downloadReport = dropTask(async () => {
+    const data = this.sortedResults.map((o) => {
+      const rhett = {};
+      rhett[this.intl.t('general.id')] = o.courseId;
+      rhett[this.intl.t('general.course')] = o.courseTitle;
+      rhett[this.intl.t('general.year')] = o.year;
+      rhett[this.intl.t('general.session')] = o.sessionTitle;
+      rhett[this.intl.t('general.firstOffering')] = o.firstOfferingDate;
+      rhett[this.intl.t('general.instructors')] = o.instructors.join(', ');
+      rhett[this.intl.t('general.learnerGroupTitle')] = o.title;
+      rhett[this.intl.t('general.link')] = o.link;
+
+      return rhett;
+    });
+    const csv = PapaParse.unparse(data);
+    this.finishedBuildingReport = true;
```

Comment:
this component property is not declared, and it appears that its value is not being processed in the template. please remove.

Reply:
Was actually a mostly implemented feature, I hooked it up. Thanks!